src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImageFunctionalTest.groovy

```diff
@@ -533,6 +551,26 @@ class DockerBuildImageFunctionalTest extends AbstractGroovyDslFunctionalTest {
             }
             
             buildTarget.finalizedBy tasks.removeImage
+
+            ${imageIdValidation()}
+        """
+    }
+
+    private static String imageIdValidation() {
+        """
+            tasks.withType(DockerBuildImage).all { Task task ->
+                def assertTask = tasks.create("assertImageIdFor\${task.name}"){
```

Comment:
Would probably make sense to upper case the first letter of the task name so that the naming sticks to camel case notation.